src/typing/typeloadModule.ml

```diff
@@ -545,13 +545,28 @@ let init_module_type ctx context_init (decl,p) =
 						ctx.m.module_types <- ((match name with None -> tsub | Some(n,pname) -> rebind tsub n pname),p) :: ctx.m.module_types
 					with Not_found ->
 						(* this might be a static property, wait later to check *)
-						let tmain = get_type tname in
-						context_init#add (fun() ->
+						let find_main_type_static () = 
+							let tmain = get_type tname in
```

Comment:
@Simn is it okay to delay this `get_type` to whenever `context_init#add` callback is called?

Reply:
oh well, seems to work